pynumaflow/sink/server.py

```diff
@@ -122,17 +120,13 @@ async def server_graceful_shutdown():
             await server.stop(5)
             """
 
-        self._cleanup_coroutines.append(server_graceful_shutdown())
+        self.cleanup_coroutines.append(server_graceful_shutdown())
         await server.wait_for_termination()
 
     def start_async(self) -> None:
         """Starts the Async gRPC server on the given UNIX socket."""
-        loop = asyncio.get_event_loop()
-        try:
-            loop.run_until_complete(self.__serve_async())
-        finally:
-            loop.run_until_complete(*self._cleanup_coroutines)
-            loop.close()
+        server = grpc.aio.server(options=self._server_options)
```

Comment:
Don't we need the Migration thread pool argument here? https://grpc.github.io/grpc/python/grpc_asyncio.html#create-server

Reply:
My understanding of Threadpool executor argument is so that asyncio can offload any blocking IO operations or calls to legacy code (read non async) on a separate thread with out blocking the event loop. In this case however, since we dont know what the exact work UDF will be doing, its not required to supply an executor here. 

for example, udf authors can always do this to execute a blocking operation in a separate thread pool


```
# create a thread pool
executor = concurrent.futures.ThreadPoolExecutor(
        max_workers=3,
    )
```


and then use the executor like so to execute blocking tasks

```
 loop = asyncio.get_event_loop()
      blocking_tasks = [
          loop.run_in_executor(executor, blocks, i)
          for i in range(6)
      ]
```
sizing the executor interms of pool size etc is totally specific to the task being executed and so i think we should let the udf authors take a call. 

I do agree that we probably need to add an example on how to do a blocking operation using asyncio.

Reply:
we can always supply a default executor, but that may not be very useful.

Reply:
one other point - ProcessPoolExecutor will provide even more isolation depending on the usecase. I believe a udf author is in the best position to pick one over the other.
wdyt?

Reply:
Thanks for the explanation, and yes I agree. Maybe we could keep it like this for now, and then add some documentation on how udf authors can manage this.